source/common/http/conn_pool_grid.cc

```diff
@@ -208,8 +212,8 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
       std::make_unique<WrapperCallbacks>(*this, decoder, pools_.begin(), callbacks);
   ConnectionPool::Cancellable* ret = wrapped_callback.get();
   LinkedList::moveIntoList(std::move(wrapped_callback), wrapped_callbacks_);
-  // Note that in the case of immediate attempt/failure, newStream will delete this.
-  if (wrapped_callbacks_.front()->newStream()) {
+  // Note that in the case of immediate attempt/failure, newStream deletes the wrapped callback.
```

Comment:
Heh, I also re-wrote this comment (differently) in my PR, by moving the text inside the if (ImmediateResult) which is nice because the comment really only applies to that branch.